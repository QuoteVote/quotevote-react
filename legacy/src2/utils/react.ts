import React, { 
  useState, 
  useEffect, 
  useRef, 
  ReactNode, 
  ChangeEvent, 
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
} from 'react';

export type { 
  ReactNode,
  ChangeEvent,
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
};

export { 
  React as default,
  useState,
  useEffect,
  useRef
};

// Add type declarations for common event handlers
export type InputChangeEvent = ChangeEvent<HTMLInputElement>;
export type TextAreaChangeEvent = ChangeEvent<HTMLTextAreaElement>;
export type ButtonClickEvent = MouseEvent<HTMLButtonElement>;
export type SetState<T> = Dispatch<SetStateAction<T>>;

// Add type declarations for common component props
export interface BaseProps {
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface InputProps extends BaseProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  disabled?: boolean;
} 