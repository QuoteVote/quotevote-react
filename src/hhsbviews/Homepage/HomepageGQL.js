import gql from 'graphql-tag'

export const ACTIVITIES_QUERY = gql`
  query activities($limit: Int!, $offset: Int!, $searchKey: String!, $activityEvent: JSON!) {
    activities(limit: $limit, offset: $offset, searchKey: $searchKey, activityEvent: $activityEvent)
  }
`
