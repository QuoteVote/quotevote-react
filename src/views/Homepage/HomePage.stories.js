import React from 'react'

import HomePage from './Homepage'

export default {
  component: HomePage,
  title: 'HomePage',
}

export const HomePageComponent = () => <HomePage />
