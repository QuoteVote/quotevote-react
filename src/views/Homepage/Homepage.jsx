import Activity from '@/components/Activity/Activity'

export default function Homepage() {
  return <Activity showSubHeader />
}
