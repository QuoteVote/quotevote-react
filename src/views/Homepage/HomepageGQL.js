import gql from 'graphql-tag'

export const ACTIVITIES_QUERY = gql`
  query activities($limit: Int!, $offset: Int!, $searchKey: String!, $startDateRange: String, $activityEvent: JSON!) {
    activities(limit: $limit, offset: $offset, searchKey: $searchKey, startDateRange: $startDateRange, activityEvent: $activityEvent)
  }
`
