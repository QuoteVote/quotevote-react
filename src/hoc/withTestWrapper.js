import React from 'react'
import { PersistGate } from 'redux-persist/integration/react'
import { ApolloProvider } from '@apollo/react-hooks'
import { Provider } from 'react-redux'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import store, { persistor } from '../store/store'
import client from '../config/apollo'
import customTheme from '../theme'
const hist = createBrowserHistory()
const theme = createTheme(customTheme)

const withTestWrapper = (Component) => (props) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider theme={theme}>
          <Router history={hist}>
            <Component {...props} />
          </Router>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  </ApolloProvider>
)
export default withTestWrapper
