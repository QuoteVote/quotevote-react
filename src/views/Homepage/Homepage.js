/* eslint-disable no-underscore-dangle */
import React, { useState } from 'react'
import GridContainer from 'mui-pro/Grid/GridContainer'
import Card from 'mui-pro/Card/Card'
import CardBody from 'mui-pro/Card/CardBody'
import CustomizedInputBase, { GET_SEARCH_KEY } from 'components/searchBar'
import Pagination from 'material-ui-flat-pagination'
import Slider from '@material-ui/core/Slider'
import Calendar from 'assets/img/Calendar.svg'
import Filter from 'assets/img/Filter.svg'
import Emoji from 'assets/img/FollowingEmoji.svg'
import AlertList from 'components/AlertList'
import { useQuery } from '@apollo/react-hooks'
import moment from 'moment'
import ToggleButton from '@material-ui/lab/ToggleButton'
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup'
import { makeStyles } from '@material-ui/core/styles'
import { ACTIVITIES_QUERY } from './HomepageGQL'

const useStyles = makeStyles({
  headerToggle: {
    fontSize: '20px',
    color: 'gray',
    font: 'League Spartan',
    fontWeight: '900',
    textDecoration: 'underline',
    textShadow: '1px 1px gray',
  },
})

const ACTIVITY_COLORS = {
  QOUTED: '#00CAE3',
  UP: '#55B559',
  DOWN: '#FF1100',
  COMMENTED: '#FF9E0F',
  HEARTED: '#E91E63',
  POSTED: '#020202',
}


function formatContentDate(sDate) {
  const a = moment.utc()
  const b = moment.utc(sDate)
  const dateDiff = a.diff(b, 'days')
  if (dateDiff <= 1) {
    return moment(sDate)
      .calendar()
      .toString()
      .replace('at', '@')
  }

  return moment(sDate).format('MMM Do')
}

export default function Homepage() {
  const classes = useStyles()
  const limit = 5
  const [offset, setOffset] = useState(1)
  const conditions = ['POSTED', 'VOTED', 'COMMENTED', 'QUOTED']
  const [selectedEvent, setSelectedEvent] = useState(conditions)
  const [selectAll, setSelectAll] = useState('ALL')
  const handleSelectAll = (event, newSelectAll) => {
    if (newSelectAll.length) {
      setSelectedEvent(conditions)
    }
    setSelectAll(newSelectAll)
  }
  const handleActivityEvent = (event, newActivityEvent) => {
    if (!newActivityEvent.length) {
      setSelectAll(['ALL'])
      setSelectedEvent(conditions)
    } else {
      const isAllToggled = newActivityEvent.length === 4
      setSelectAll(isAllToggled ? ['ALL'] : [])
      setSelectedEvent(newActivityEvent)
    }
  }
  const [total, setTotal] = useState(1)

  const handleSlider = (event, newValue) => {
    setOffset(newValue)
  }

  const { data: { searchKey } } = useQuery(GET_SEARCH_KEY)
  const { loading, data } = useQuery(ACTIVITIES_QUERY, {
    variables: {
      limit, offset, searchKey, activityEvent: selectedEvent,
    },
  })

  const { activities } = (!loading && data && data.activities) || { activities: { activities: [], total: 0 } }
  React.useEffect(() => {
    if (data) {
      setTotal(data.activities.total)
    }
  }, [data])

  const activitiesData = !loading && activities && activities.length && activities.map((activity) => {
    const time = activity && formatContentDate(activity.data.created)
    switch (activity.event) {
      case 'VOTED':
        return {
          id: activity.data._id,
          AlertTitle: `${activity.data.type.toUpperCase()}VOTED`,
          color: ACTIVITY_COLORS[`${activity.data.type.toUpperCase()}`],
          AlertBody: activity.data.content.title,
          time,
          points: activity.data.type === 'up' ? `+${activity.data.points}` : `-${activity.data.points}`,
          creator: activity.data.creator,
        }
      case 'POSTED':
        return {
          id: activity.data._id,
          AlertTitle: 'CONTENT',
          color: ACTIVITY_COLORS.POSTED,
          AlertBody: activity.data.title,
          time,
          points: '',
          creator: activity.data.creator,
        }
      case 'QUOTED':
        return {
          id: activity.data._id,
          AlertTitle: activity.event,
          color: ACTIVITY_COLORS.QOUTED,
          AlertBody: `"${activity.data.quote}"`,
          time,
          points: '',
          creator: activity.data.creator,
        }
      case 'COMMENTED':
        return {
          id: activity.data._id,
          AlertTitle: activity.event,
          color: ACTIVITY_COLORS.COMMENTED,
          AlertBody: `"${activity.data.content}"`,
          time,
          points: '',
          creator: activity.data.creator,
        }
      case 'HEARTED':
        return {
          id: activity.data._id,
          AlertTitle: activity.event,
          color: ACTIVITY_COLORS.HEARTED,
          AlertBody: activity.data.content.title,
          time,
          points: '',
          creator: activity.data.creator,
        }
      default:
        break
    }
    return null
  })

  return (
    <Card style={{ display: 'flex', flexBasis: '800px' }}>
      <CardBody>
        <GridContainer
          direction="row"
          justify="center"
          alignItems="center"

        >
          <GridContainer alignItems="center" direction="row" style={{ width: '50%' }}>
            <GridContainer justify="center" wrap="nowrap" direction="row">
              <p>
                <ToggleButtonGroup value={selectAll} onChange={handleSelectAll} aria-label="All Event">
                  <ToggleButton
                    value="ALL"
                    aria-label="All"
                    className={classes.headerToggle}
                  >
                    All
                  </ToggleButton>
                </ToggleButtonGroup>
                <ToggleButtonGroup value={selectedEvent} onChange={handleActivityEvent} aria-label="Event">
                  <ToggleButton
                    value="POSTED"
                    aria-label="Content"
                    className={classes.headerToggle}
                  >
                    Content
                  </ToggleButton>
                  <ToggleButton
                    value="VOTED"
                    aria-label="phone"
                    className={classes.headerToggle}
                  >
                    Votes
                  </ToggleButton>
                  <ToggleButton
                    value="COMMENTED"
                    aria-label="phone"
                    className={classes.headerToggle}
                  >
                    Comments
                  </ToggleButton>
                  <ToggleButton
                    value="QUOTED"
                    aria-label="phone"
                    className={classes.headerToggle}
                  >
                    Quotes
                  </ToggleButton>
                </ToggleButtonGroup>
              </p>

            </GridContainer>
            <Slider
              defaultValue={limit}
              value={offset}
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              max={total}
              min={1}
              onChange={handleSlider}
            />

          </GridContainer>
          <br></br>
          <br></br>

          <GridContainer
            alignItems="center"
            direction="row"
            justify="space-between"
            style={{
              backgroundColor: '#FF7A00',
              boxShadow: '0 6px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)',
              width: '75%',
              wrap: 'nowrap',
            }}
          >
            <h3
              style={{
                color: 'white',
                font: 'League Spartan',
                fontWeight: 'bold',
                paddingLeft: '20px',
                paddingBottom: '5px',
              }}
            >
              Activity Feed
            </h3>

            <div style={{ display: 'flex', justifyContent: 'flex-end', flexDirection: 'row' }}>
              <CustomizedInputBase setOffset={setOffset} />
              <img
                alt="Calendar Icon"
                src={Calendar}
                style={{ display: 'flex', maxHeight: '40px', paddingLeft: '15px' }}
              />
              <img alt="Filter Icon" src={Filter} style={{ display: 'flex', maxHeight: '40px', paddingLeft: '15px' }} />
              <img
                alt="Emoji Icon"
                src={Emoji}
                style={{
                  display: 'flex', maxHeight: '40px', paddingLeft: '15px', paddingRight: '15px',
                }}
              />
            </div>

          </GridContainer>

        </GridContainer>
        <br></br>
        <br></br>
        <AlertList Data={activitiesData} loading={loading} limit={limit} />
      </CardBody>
      <Pagination
        style={{ margin: 'auto' }}
        limit={limit}
        offset={offset}
        total={total}
        onClick={(e, offsetVal) => setOffset(offsetVal)}
      />
    </Card>


  )
}
