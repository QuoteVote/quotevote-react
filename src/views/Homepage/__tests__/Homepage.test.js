import React from 'react'
import { render } from '@testing-library/react'
import Homepage from '../Homepage'
import withTestWrapper from '../../../hoc/withTestWrapper'

const HomepageWrapper = withTestWrapper(Homepage)

describe('Post test -', () => {
  it('renders correctly', () => {
    const { container } = render(
      <HomepageWrapper />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
