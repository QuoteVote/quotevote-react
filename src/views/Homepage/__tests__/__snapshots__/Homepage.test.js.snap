// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post test - renders correctly 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root makeStyles-header-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 MuiGrid-grid-md-3"
      >
        <p
          class="MuiTypography-root makeStyles-headerName-3 MuiTypography-body1"
        >
          Activity Feed
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-6"
      >
        <div
          class="MuiPaper-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-6"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiInputBase-root makeStyles-input-5 MuiInputBase-fullWidth"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-6"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 MuiGrid-grid-md-3"
      />
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  />
</div>
`;
